sdk/cosmosdb/cosmos/src/retry/retryUtility.ts

```diff
@@ -36,92 +42,111 @@ interface RetryPolicies {
  * @hidden
  */
 export async function execute({
+  diagnosticNode,
   retryContext = { retryCount: 0 },
   retryPolicies,
   requestContext,
   executeRequest,
 }: ExecuteArgs): Promise<Response<any>> {
   // TODO: any response
-  if (!retryPolicies) {
-    retryPolicies = {
-      endpointDiscoveryRetryPolicy: new EndpointDiscoveryRetryPolicy(
-        requestContext.globalEndpointManager,
-        requestContext.operationType
-      ),
-      resourceThrottleRetryPolicy: new ResourceThrottleRetryPolicy(
-        requestContext.connectionPolicy.retryOptions.maxRetryAttemptCount,
-        requestContext.connectionPolicy.retryOptions.fixedRetryIntervalInMilliseconds,
-        requestContext.connectionPolicy.retryOptions.maxWaitTimeInSeconds
-      ),
-      sessionReadRetryPolicy: new SessionRetryPolicy(
-        requestContext.globalEndpointManager,
+  return addDignosticChild(
```

Comment:
@v1k1 if I understand it correctly, does this mean that if we have infinite retrials then there would be infinite numbers of nodes getting created?

Reply:
Yes, but only if the `CosmosDBDiagnostic` level is in debug or debug-unsafe mode.

Reply:
we do have certain cases where we do retry for >1000 times. Should we do a perf test just to be on safe side?